Kernel/FileSystem/VirtualFileSystem.cpp

```diff
@@ -720,12 +720,15 @@ KResultOr<NonnullRefPtr<Custody>> VFS::resolve_path(StringView path, Custody& ba
             if (symlink_target.is_error())
                 return symlink_target;
 
-            bool have_more_parts = i + 1 < parts.size();
-            if (i + 1 == parts.size() - 1 && parts[i + 1].is_empty())
-                have_more_parts = false;
+            if (i == parts.size() - 1)
+                return symlink_target;
+
+            if (i == parts.size() - 2 && parts[i + 1].is_empty())
+                return symlink_target;
 
-            if (!have_more_parts)
+            if (parts[i + 1].is_null()) {
```

Comment:
I'm not sure whether it's correct to just skip the part if it is null, or if it is just incorrect for it to be null in the first place...